src/main/java/com/thebois/models/world/structures/AbstractStructure.java

```diff
@@ -61,4 +66,62 @@ public StructureType getType() {
         return structureType;
     }
 
+    @Override
+    public Collection<IItem> neededItems() {
+        // Make a copy of all IItems in collectionOfAllNeededItems
+        final Collection<IItem> itemDifference = new ArrayList<>(allNeededItems);
+        // Remove IItems that have been delivered
+        deliveredItems.forEach(itemDifference::remove);
+        // Return difference of the two collections
+        return itemDifference;
+    }
+
+    @Override
+    public boolean deliverItem(final IItem deliveredItem) {
+        if (neededItems().contains(deliveredItem)) {
+            deliveredItems.add(deliveredItem);
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    @Override
+    public float builtStatus() {
```

Comment:
style

Should be verb-ified to something like `getBuiltStatus` or `calculateBuiltStatus`.